test/parallel/test-fs-chmod-mask.js

```diff
@@ -71,23 +71,23 @@ function test(mode, asString) {
   if (fs.lchmod) {
     const link = path.join(tmpdir.path, `lchmod-src-${suffix}`);
     const file = path.join(tmpdir.path, `lchmod-dest-${suffix}`);
-    fs.writeFileSync(file, 'test', 'utf-8');
+    fs.writeFileSync(file, 'test', { encoding: 'utf-8', mode: 0o444 });
     fs.symlinkSync(file, link);
 
     fs.lchmod(link, input, common.mustCall((err) => {
       assert.ifError(err);
-      assert.strictEqual(fs.lstatSync(link).mode & 0o777, mode);
+      assert.strictEqual(fs.lstatSync(link).mode & 0o444, 0o444);
```

Comment:
1. IIUC `mode` is an argument so I don't think it should be converted to a constant.
2. This seems to indicate that the change is limited to changes in the following bits ‭100'100'100‬ (0o444) while before the test had wider range 111'111'111 (0o777)? If this is correct it make this a semver-major change, and I'm not sure that what we want.

Reply:
Hey @refack thanks for taking time to review this. 
I was trying to add a regression test here (to fails if changes in `fs.js` are reverted). The previous assertions passes regardless of the changes I made. However, this takes care of test failing in case of regression.
Let me know if I need to implement some other work around to add a regression test.

Reply:
> The previous assertions passes regardless of the changes I made. However, this takes care of test failing in case of regression.

That fantastic. So simply duplicate the tests (keep the old ones, and add these).
IMHO you should define `const kMaskOnlyRead = 0x444`
and in your new (duplicated) tests do:

```suggestion
      assert.strictEqual(fs.lstatSync(link).mode & kMaskOnlyRead, mode & kMaskOnlyRead);
```